dapr/clients/grpc/_request.py

```diff
@@ -5,31 +5,89 @@
 Licensed under the MIT License.
 """
 
-from typing import Optional, Union, Dict
+from typing import Dict, Optional, Union
 
-from dapr.clients.grpc._helpers import MetadataTuple
 from google.protobuf.any_pb2 import Any as GrpcAny
 from google.protobuf.message import Message as GrpcMessage
 
 from dapr.clients.base import DEFAULT_JSON_CONTENT_TYPE
+from dapr.clients.grpc._helpers import (
+    MetadataDict,
+    MetadataTuple,
+    tuple_to_dict,
+    to_bytes,
+    to_str,
+    unpack
+)
 
 
-class InvokeServiceRequestData:
+class DaprRequest:
+    """A base class for Dapr Request.
+
+    This is the base class for Dapr Request. User can gets the metadata.
+
+    Attributes:
+        metadata(dict): A dict to include the headers from Dapr Request.
+    """
+    def __init__(self, metadata: MetadataTuple = ()):
+        self.metadata = metadata  # type: ignore
+
+    @property
+    def metadata(self) -> MetadataDict:
+        """Get metadata from :class:`DaprRequest`."""
+        return self.get_metadata(as_dict=True)  # type: ignore
+
+    @metadata.setter
+    def metadata(self, val) -> None:
```

Comment:
type hint? 
shouldn't the type check be `isinstance(val, MetadataTuple) ?

Reply:
isinstance is working only for python primitive type :) TypeHinting != pritmitive type.